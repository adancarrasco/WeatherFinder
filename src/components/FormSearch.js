import React, { Component } from "react";

import { fetchWeatherByCityName } from "../utils/weatherSvs";

import styles from "./FormSearch.module.css";

class FormSearch extends Component {
  constructor(props) {
    super(props);
    this.state = {
      cityName: ""
    };
  }

  /**
   * Fetches the weather info; if response.status is 200 fills the state
   * with the weather info, otherwise handles the 404 error or
   * catch a different one
   */
  getWeatherInfo = async cityName => {
    let cityWeatherInfo;
    try {
      const response = await fetchWeatherByCityName(cityName);
      if (!response || response.status !== 200) {
        throw new Error();
      }
      cityWeatherInfo = response.data;
      return { ...cityWeatherInfo, cityName };
    } catch (error) {
      console.error(error);
      return { cityName, error: true };
    }
  };

  handleChange = event => {
    this.setState({ cityName: event.target.value });
  };

  searchWeather = async cityName => {
    const cityWeatherInfo = await this.getWeatherInfo(cityName);
    this.props.handleSearch(cityWeatherInfo);
  }

  handleSubmit = event => {
    event.preventDefault();
    this.searchWeather(this.state.cityName);
  };

  render() {
    return (
      <div className={styles.searchFormContainer}>
        <form className={styles.searchForm} onSubmit={this.handleSubmit}>
          <input
            className={styles.inputSearch}
            type="text"
            placeholder="Input the city name"
            onChange={this.handleChange}
            value={this.state.cityName}
          />
          <input className={styles.searchButton} type="submit" value="Search" />
        </form>
      </div>
    );
  }
}

export default FormSearch;
