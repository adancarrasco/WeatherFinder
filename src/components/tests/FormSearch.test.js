import React from "react";
import { shallow } from "enzyme";

import FormSearch from "../FormSearch";

describe("FormSearch", () => {
  it("it renders correctly", () => {
    const wrapper = shallow(<FormSearch />);
    expect(wrapper).toMatchSnapshot();
  });
});
