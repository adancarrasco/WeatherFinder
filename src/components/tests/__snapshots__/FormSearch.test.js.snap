// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSearch it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormSearch />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="searchForm"
        onSubmit={[Function]}
      >
        <input
          className="inputSearch"
          onChange={[Function]}
          placeholder="Input the city name"
          type="text"
          value=""
        />
        <input
          className="searchButton"
          type="submit"
          value="Search"
        />
      </form>,
      "className": "searchFormContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="inputSearch"
            onChange={[Function]}
            placeholder="Input the city name"
            type="text"
            value=""
          />,
          <input
            className="searchButton"
            type="submit"
            value="Search"
          />,
        ],
        "className": "searchForm",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "inputSearch",
            "onChange": [Function],
            "placeholder": "Input the city name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "searchButton",
            "type": "submit",
            "value": "Search",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="searchForm"
          onSubmit={[Function]}
        >
          <input
            className="inputSearch"
            onChange={[Function]}
            placeholder="Input the city name"
            type="text"
            value=""
          />
          <input
            className="searchButton"
            type="submit"
            value="Search"
          />
        </form>,
        "className": "searchFormContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="inputSearch"
              onChange={[Function]}
              placeholder="Input the city name"
              type="text"
              value=""
            />,
            <input
              className="searchButton"
              type="submit"
              value="Search"
            />,
          ],
          "className": "searchForm",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "inputSearch",
              "onChange": [Function],
              "placeholder": "Input the city name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "searchButton",
              "type": "submit",
              "value": "Search",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
